### PR TITLE
Group Normalization

### PR DESCRIPTION
**Pull request by zhanghang1989**
## Description ##
Using the BN as the backend. Currently, the block is not hybridizable, due to reading batch size.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by roywei** [comment]
@mxnet-label-bot add[Gluon, pr-awaiting-review]

File: python/mxnet/gluon/nn/basic_layers.py [hunk 1]
```diff
@@ -704,3 +706,82 @@ def hybrid_forward(self, F, x, *args):
...
                                            function=self._func_name)
+
+
+class GroupNorm(HybridBlock):
+    """GroupNorm normalization layer (Wu and He, 2014).
+
+    Parameters
+    ----------
+    ngroups : int
+        Numnber of channel groups in GN.
```
Comment (wkcn) [inline python/mxnet/gluon/nn/basic_layers.py]: Thanks for your contribution!
There is a typo 'numnber'.

File: tests/python/unittest/test_gluon.py [hunk 2]
```diff
@@ -2587,6 +2587,21 @@ def hybrid_forward(self, F, x):
...
+
+@with_seed()
+def test_groupnorm():
+    ctx=mx.context.current_context()
+    x = mx.nd.random.uniform(1, 2, (4, 16, 8, 8), ctx=ctx)
+    gn = mx.gluon.nn.GroupNorm(4, 16)
+    gn.initialize(ctx=ctx)
+    y = gn(x)
+    y = y.reshape(0, 4, -1)
+    print('y.mean(2) =', y.mean(2))
```
Comment (anirudhacharya) [inline tests/python/unittest/test_gluon.py]: can we use logging.info instead of a print statement here?

Comment (zhanghang1989) [reply tests/python/unittest/test_gluon.py]: thanks for pointing out :)

**Comment by karan6181** [comment]
@zhanghang1989 Can you please address the review comments and try to merge this PR? Thanks!

**Comment by abhinavs95** [comment]
@zhanghang1989 Could you please provide an update on the PR? Thanks

**Comment by zhanghang1989** [comment]
Thanks for following up. We temporarily move it to gluon-cv.

**Pull request closed without merging**